docs/usage/key-concepts/how-it-works.md

```diff
@@ -0,0 +1,152 @@
+---
+title: How it works ?
+description: Learn all about Renovate's workflow
+---
+
+# Introduction
+
+Renovate works by searching for all the dependencies in your project and then look for updates for each.
+
+To be compatible with the variety of dependency-naming and versioning conventions, Renovate has defined modules for each known convention, and you can define your own too.
+
+Please add comments to the [issue#25091](https://github.com/renovatebot/renovate/issues/25091) if you wish to see a part (better) shown in the graph below
+
+## Modules
+
+The three modules are: [manager](../modules/manager/index.md), [datasource](../modules/datasource/index.md) and [versioning](../modules/versioning.md).
+
+They are used sequentially:
+
+1. the manager module looks for files based on their name and extract dependencies from them, each dependency has a datasource
+2. the datasource module looks for the existing versions of the dependency
+3. the versioning module search for a valid version regarding the dependency's version
+
+For example:
+
+1. the `gitlabci` manager finds a dependency named `python:3.10-alpine` of datasource `docker`
+2. the `docker` datasource looks for versions and finds `[python:3.9,python:3.9-alpine,python:3.10,python:3.10-alpine,python:3.11,python:3.11-alpine]`
+3. the `docker` versioning takes `python:3.11-alpine` as it is compatible with `python:3.10-alpine`
+
+# Workflow
+
+## Basic
+
+Here's a high-level overview of the renovate workflow, collect dependencies then update them:
+
+```mermaid
+flowchart LR
+  subgraph COLLECT
+    direction TB
+    CC[[For each manager]]
+    CC -->|managerA| CD["..."]
+    CC -->|managerB| CCF["collect files"]
+
+    CCF --> CFEF[[For each file]]
+
+    CFEF -->|file1| CCD1[Collect dependency]
+    CFEF -->|file2| CCD2[...]
+  end
+
+  subgraph UPDATE
+    direction TB
+
+    UC[[For each manager]]
+    UC -->|managerA| UD["..."]
+    UC -->|managerB| UFEF[[For each file]]
+
+    UFEF -->|file1| FED[[For each dependency]]
+    UFEF -->|file2| FED2[...]
+
+
+    FED -->|dep1| D1[...]
+    D1 -..-> CU
+    FED -->|dep2| D2[use datasource to\n get possible updates]
+    D2 --> J[use versionning to find \n next valid update]
+
+    FED2 -...-> CU
+
+    UD -....-> CU
+    J --> CU[Collect updates]
+
+    CU --> FEU[[For each update]]
+
+    FEU --> AU[Create branch\nApply update\nSetup PR]
+
+  end
+
+  COLLECT --> UPDATE
+```
+
+## Advanced
+
+And here's a mode detailed view of the workflow:
+
+```mermaid
+flowchart TB
+    subgraph INITIALIZATION
+        direction TB
+        MC[Merge configuration sources \n most important to least: \n cli > env > file > default]
```

Comment:
I don't know what "merge configuration sources" means.

Reply:
It means : merging the condigurtion that comes from the cli, the env, the given config file and finally the default renovate's config